### DISCUSSION TITLE
Prometheus retention time

### DISCUSSION THREAD

**Discussion by korontanis** [carlosedp/cluster-monitoring#167] [Q&A]
Hi,

I have Prometheus retention time to 5 days. I have created a custom component that performs custom defined queries on prometheus periodically. These custom created queries are defined on prometheus-rules.yaml file.  It seems that after 5 days it is not possible to query container metrics of cadvisor. Do you know if there is a problem with Prometheus v2.19 or if it is something else that cause this problem? The weird thing is that my component normally retrieves the metrics of node exporter. Could the problem be related with Kubernetes configurations?

**Comment by carlosedp** [comment]
From what I understood, you configured Prometheus to purge data after 5 days so you won't be able to query data older than this.
The elements that are queried normally don't store data.
